### PR TITLE
feat: Implement worker profile sliding window

### PR DESCRIPTION
**Pull request by AnotherMedo**
Following the figma, clicking `book` on a worker card should show a sliding window displaying the worker information, reviews and segway to the next step in ordering a **_QuickFix_**.

I will implement this feature in three steps:
- Add an element for the sliding window. This is important as this element is predicted to be reused in the message screen when a _**QuickFix**_ is requested by a user.
- Add a scaffold of the sliding window displaying properly¹
- Populate this window with all the relevant information²

**Commit History:**
`feat: add sliding window element`
- [X] add the sliding window in `com.arygm.quickfix.ui.elements`
- [X] visually test the sliding window with a test composable

`feat: add a sliding window to SearchWorkerResult`
- [x] add a sliding window that pops up when book is clicked on a worker card

`feat: populate the sliding window as per the figma`
- [x] populate sliding window with relevant information

`test: QuickFixSlidingWindow and ktfmt format`
`test: add tests for sliding window functionality`
- [x] test `QuickFixScrollingWindow.kt`
- [x] add tests to `SearchWorkerResult.kt`³

`chore: format files as per ktfmt`
- [x] `ktfmt` formatting
- [x] general debug and miscellaneous

###### ³: Thanks to @Sebbayra for helping me figure out the tests! 😄

### CODE REVIEW

File: app/src/androidTest/java/com/arygm/quickfix/ui/search/SearchWorkerResultScreenTest.kt
Comment (Sebbayra) [inline app/src/androidTest/java/com/arygm/quickfix/ui/search/SearchWorkerResultScreenTest.kt:1]: Tests are great, glad i was able to help out a bit.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSlidingWindow.kt
Comment (Sebbayra) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSlidingWindow.kt:1]: Design and implementation are perfect we should just open an issue about the bottom bar visibility when we show the sliding window.

File: app/src/main/java/com/arygm/quickfix/ui/search/SearchWorkerResult.kt [hunk 1]
```diff
@@ -90,158 +117,558 @@ val listOfButtons =
...
+          "Cabinet Painting",
+          "Licensed & Insured",
+          "Local Worker")
+  val reviews =
+      listOf(
+          "Overall, the work was shit; it’s fair to say that since he just painted the whole mural with " +
+              "his shit. I don’t know what to do; I’ll probably sue him.",
+          "Moha was very punctual and did a great job painting our living room. Highly recommended!",
+          "Moha was very professional and did a fantastic job painting our house",
+          "I wanna marry that man,")
```
Comment (Sebbayra) [inline app/src/main/java/com/arygm/quickfix/ui/search/SearchWorkerResult.kt:186]: Again not much to talk about, mehdi is goated but we should remove these eventually and adapt the class with the quickfix backend (I just pushed a PR about it)

**Review by Sebbayra — APPROVED** [review]
Wonderful work, just don't forget to open the issue about the bottom nav bar.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSlidingWindow.kt
Comment (daferya) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSlidingWindow.kt:1]: This is a solid implementation with clean design and functionality. Handling edge cases will make it even better.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSlidingWindow.kt [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                              coroutineScope.launch {
+                                animatableOffsetX.animateTo(
+                                    targetValue = gapWidthPx,
+                                    animationSpec = tween(durationMillis = 100))
+                              }
+                              dragOffsetX = 0f
+                            }
+                          },
+                          onHorizontalDrag = { _, dragAmount ->
+                            dragOffsetX = (dragOffsetX + dragAmount).coerceAtLeast(0f)
```
Comment (daferya) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSlidingWindow.kt:104]: you can add a minDragDistance to ignore tiny, unintentional drags

Comment (AnotherMedo) [reply app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSlidingWindow.kt:104]: it already handles this 😄

**Pull request merged**